starfish/core/spots/DecodeSpots/util.py

```diff
@@ -0,0 +1,166 @@
+    ind[anchor_round] = np.arange(reference_df.shape[0], dtype=np.int32)
+
+    # get spots matching across rounds
+    for r in sorted(set(round_dataframes.keys()) - {anchor_round, }):
+        query_df = round_dataframes[r]
+        query_coordinates = query_df[[Axes.ZPLANE, Axes.Y, Axes.X]]
+        nn = NearestNeighbors(n_neighbors=1)
+        nn.fit(query_coordinates)
+        distances, indices = nn.kneighbors(reference_coordinates)
+        dist[r] = distances
+        ind[r] = indices
+
+    return dist, ind
+
+
+def _build_intensity_table(
+    round_dataframes: Dict[int, pd.DataFrame],
+    dist: pd.DataFrame,
+    ind: pd.DataFrame,
+    channels: Sequence[int],
+    rounds: Sequence[int],
+    search_radius: int,
+    anchor_round: int,
+) -> IntensityTable:
+    """Construct an intensity table from the results of a local search over detected spots
+
+    Parameters
+    ----------
+    round_dataframes : Dict[int, pd.DataFrame]
+        Output from _merge_spots_by_round, contains mapping of image volumes from each round to
+        all the spots detected in them.
+    dist, ind : pd.DataFrame
+        Output from _match_spots, contains distances and indices to the nearest spot for each
+        spot in anchor_round.
+    channels, rounds : Sequence[int]
+        Channels and rounds present in the ImageStack from which spots were detected.
+    search_radius : int
+        The maximum (euclidean) distance in pixels for a spot to be considered matching in
+        a round subsequent to the anchor round.
+    anchor_round : int
+        The imaging round to seed the local search from.
+
+    """
+
+    anchor_df = round_dataframes[anchor_round]
+
+    # create empty IntensityTable filled with np.nan
+    data = np.full((dist.shape[0], len(channels), len(rounds)), fill_value=np.nan)
+    dims = (Features.AXIS, Axes.CH.value, Axes.ROUND.value)
+    coords: Mapping[Hashable, Tuple[str, Any]] = {
+        Features.SPOT_RADIUS: (Features.AXIS, anchor_df[Features.SPOT_RADIUS]),
+        Axes.ZPLANE.value: (Features.AXIS, anchor_df[Axes.ZPLANE]),
+        Axes.Y.value: (Features.AXIS, anchor_df[Axes.Y]),
+        Axes.X.value: (Features.AXIS, anchor_df[Axes.X]),
+        Axes.ROUND.value: (Axes.ROUND.value, rounds),
+        Axes.CH.value: (Axes.CH.value, channels)
+    }
+    intensity_table = IntensityTable(data=data, dims=dims, coords=coords)
+
+    # fill IntensityTable
+    for r in rounds:
+        # get intensity data and indices
+        spot_indices = ind[r]
+        intensity_data = round_dataframes[r].loc[spot_indices, Features.INTENSITY]
+        channel_index = round_dataframes[r].loc[spot_indices, Axes.CH]
+        round_index = np.full(ind.shape[0], fill_value=r, dtype=int)
+        feature_index = np.arange(ind.shape[0], dtype=int)
+
+        # mask spots that are outside the search radius
+        mask = np.asarray(dist[r] < search_radius)  # indices need not match
```

Comment:
that's a good question @ambrosejcarr ?

Reply:
Given the properties of the data acquisition, In most cases it should not be treated the same (although there are exceptions). This is the way the search was implemented by the authors of the pipeline, and so it is how we implemented it. If it is simple to set search radii differently for x, y, z, that would be an improvement.

Reply:
that's a really good question

Reply:
I propose that we file an issue to discuss this, and perhaps engage some ppl who use this approach to discuss.  It may not matter in the end, but it does not currently pass the smell test. :)